src/utils/pgwire/src/pg_server.rs

```diff
@@ -155,17 +158,81 @@ pub enum UserAuthenticator {
         encrypted_password: Vec<u8>,
         salt: [u8; 4],
     },
+    OAuth(HashMap<String, String>),
+}
+
+#[derive(Debug, Deserialize)]
+struct Jwks {
+    keys: Vec<Jwk>,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Deserialize)]
+struct Jwk {
+    kid: String,
+    alg: String,
+    n: String,
+    e: String,
+}
```

Comment:
Would you mind adding some doc string for these structs and fields? Or leave some links to say the RFC page. So that other developers who are not familiar with OAuth can refer to them conveniently.